nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -657,6 +653,25 @@ in {
   };
 
   config = mkIf cfg.enable {
+    assertions = [
+      { assertion = hasLocalPostgresDB -> config.services.postgresql.enable;
```

Comment:
The only remark: If a config has `postgresql.enable = true` for another service, this message will not be shown.
Only a check of the system logs will point to the missing database. We cannot check for everything :woman_shrugging:

Reply:
Well, one of the reasons for this assertion is to ensure that people with an existing matrix-synapse and a local database don't deploy a config without a postgresql service (that isn't enabled anymore by this module).